http-client-spi/src/main/java/software/amazon/awssdk/http/SdkHttpConfigurationOption.java

```diff
@@ -30,25 +30,42 @@
  */
 @SdkProtectedApi
 public final class SdkHttpConfigurationOption<T> extends AttributeMap.Key<T> {
-
     /**
      * Timeout for each read to the underlying socket.
      */
-    public static final SdkHttpConfigurationOption<Duration> SOCKET_TIMEOUT =
-            new SdkHttpConfigurationOption<>("SocketTimeout", Duration.class);
+    public static final SdkHttpConfigurationOption<Duration> SOCKET_READ_TIMEOUT =
```

Comment:
Maybe rename these to WRITE_TIMEOUT and READ_TIMEOUT. Netty for example times out when no messages have been received by that idle state handler so it doesn't exactly match up with a socket timeout.